File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

void draw(int height, int h);

int main(void)
{
    int size; 
    do
    {
        size = get_int("How tall do you want your pyramid to be?:");
    }
    while (size < 1 || size > 8);
    
    draw(size,size);
}

void draw(int height, int h)
{
    if (h == 0)
    {
        return;
    }
    
    draw(height, h-1);

    for (int n = h; n < height; n++)
    {
        printf(" ");
    }
    for (int m = (height - h); m < height; m++)
    {
        printf("#");
    }
    
    printf("  ");

    for (int i = 0; i < h; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: pset6/mario/mario.py
from cs50 import get_int

def main():
	while True:
		height = get_int("Height: ")
		if 0<height<9:
			break

	draw_triangle(height)

def draw_triangle(height):
	count = height
	for i in range(height):
		for a in range(count - i - 1):
			print(" ", end="")

		for b in range(i + 1):
			print("#", end="")

		print("  ", end="")

		for c in range(i + 1): 
			print("#", end="")

		print()

main()


